### PR TITLE
Delete PVC once the VolumeSnapshot is ready to use

### PR DESCRIPTION
**Pull request by jenting**
## Description

Directly delete the workspace Pod and remove the finalizer before the VolumeSnapshot is ready, the VolumeSnapshot is created, and the workspace Pod is removed as well. However, the PVC does not be removed after that.

## Related Issue(s)

Partially #10531

## How to test

1. Create a workspace-preview environment
2. Enable feature flag `persistent_volume_claim`.
3. Launch a workspace.
4. Manually delete the workspace pod `kubectl delete pod <pod-name>`.
5. Before the VolumeSnapshot is ready to use, remove the workspace pod finalizer `kubectl patch pod <pod-name> -p '{"metadata":{"finalizers":null}}' --type=merge`.

## Release Notes

```release-note
None
```

## Documentation

None

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

File: components/ws-manager/pkg/manager/annotations.go [hunk 1]
```diff
@@ -108,6 +108,9 @@ func (m *Manager) markWorkspace(ctx context.Context, workspaceID string, annotat
 	err := retry.RetryOnConflict(backoff, func() error {
 		pod, err := m.findWorkspacePod(ctx, workspaceID)
 		if err != nil {
+			if isKubernetesObjNotFoundError(err) {
+				return nil
+			}
 			return fmt.Errorf("cannot find workspace %s: %w", workspaceID, err)
```
Comment (sagor999) [inline components/ws-manager/pkg/manager/annotations.go:114]: if we filter not found error above, then error message below no longer makes sense.
```suggestion
			return fmt.Errorf("cannot mark workspace %s: %w", workspaceID, err)
```

File: components/ws-manager/pkg/manager/monitor.go [hunk 2]
```diff
@@ -183,19 +183,20 @@ func (m *Monitor) onVolumesnapshotEvent(evt watch.Event) error {
 
 	// get the pod resource
 	var pod corev1.Pod
-	if err := m.manager.Clientset.Get(context.Background(), types.NamespacedName{Namespace: vs.Namespace, Name: podName}, &pod); err != nil {
-		log.WithError(err).Warnf("the pod %s/%s is missing", podName, vs.Namespace)
-		return nil
-	}
+	_ = m.manager.Clientset.Get(context.Background(), types.NamespacedName{Namespace: vs.Namespace, Name: podName}, &pod)
```
Comment (sagor999) [inline components/ws-manager/pkg/manager/monitor.go:186]: hm. should we at least log any other error that might happen here except for not found error?

**Review by sagor999 — APPROVED** [review]
/hold
for minor nits.
otherwise LGTM

File: components/ws-manager/pkg/manager/annotations.go (hunk 1 above)
Comment (jenting) [reply components/ws-manager/pkg/manager/annotations.go]: fixed

File: components/ws-manager/pkg/manager/monitor.go (hunk 2 above)
Comment (jenting) [reply components/ws-manager/pkg/manager/monitor.go]: fixed

**Pull request merged**